index.d.ts

```diff
@@ -42,17 +42,55 @@ declare class Agent implements Taggable, StartSpanFn {
   currentTraceparent: string | null;
 
   // Transactions
-  startTransaction (
+  startTransaction(
     name?: string,
-    type?: string,
+    options?: TransactionOptions
+  ): Transaction | null;
+  startTransaction(
+    name: string,
+    type: string,
+    options?: TransactionOptions
+  ): Transaction | null;
+  startTransaction(
+    name: string,
+    type: string,
+    subtype: string,
+    options?: TransactionOptions
+  ): Transaction | null;
+  startTransaction(
+    name: string,
+    type: string,
+    subtype: string,
+    action: string,
     options?: TransactionOptions
   ): Transaction | null;
   setTransactionName (name: string): void;
   endTransaction (result?: string | number, endTime?: number): void;
   currentTransaction: Transaction | null;
 
   // Spans
-  startSpan (name?: string, type?: string, options?: SpanOptions): Span | null;
+  startSpan(
+    name?: string,
```

Comment:
In pure JavaScript we currently allow the user to start a span without providing a name, e.g. `startSpan(null, { ... })`. To allow this in TypeScript Typings, we should change it to:

```suggestion
    name?: string | null,
```

Reply:
fixed